api/jobset/v1alpha2/jobset_webhook.go

```diff
@@ -91,6 +91,11 @@ func (js *JobSet) ValidateCreate() (admission.Warnings, error) {
 		for _, errMessage := range validation.IsDNS1123Subdomain(js.Spec.Network.Subdomain) {
 			allErrs = append(allErrs, fmt.Errorf(errMessage))
 		}
+
+		// Since subdomain name is also used as service name, it must adhere to RFC 1035 as well.
+		for _, errMessage := range validation.IsDNS1035Label(js.Spec.Network.Subdomain) {
```

Comment:
the subdomain field in the podspec is validated at: https://github.com/kubernetes/kubernetes/blob/6c39a3724d5ff33564d31d5097704f15852c036f/pkg/apis/core/validation/validation.go#L3990

we should do the same here

Reply:
ah, you are trying to validate for service too, sgtm.

/lgtm